pkg/controller/util.go

```diff
@@ -1188,11 +1188,18 @@ func handleFailedPod(err error, podName string, pvc *v1.PersistentVolumeClaim, r
 	// Error handling to fine-tune the event with pertinent info
 	if errQuotaExceeded(err) {
 		reason = ErrExceededQuota
-	} // TODO: Add more error cases
+	}
 
 	recorder.Event(pvc, v1.EventTypeWarning, reason, msg)
 
+	if isCloneSourcePod := createCloneSourcePodName(pvc) == podName; isCloneSourcePod {
+		AddAnnotation(pvc, AnnSourceRunningCondition, "false")
+		AddAnnotation(pvc, AnnSourceRunningConditionReason, reason)
+		AddAnnotation(pvc, AnnSourceRunningConditionMessage, msg)
+	}
 	AddAnnotation(pvc, AnnRunningCondition, "false")
+	AddAnnotation(pvc, AnnRunningConditionReason, reason)
```

Comment:
Should this not be in an else.
```
if isCloneSourcePod {
  // AddCloneSourceAnnotation
} else {
  // AddTargetAnnotation
}
```
This way the source pod will not influence the target annotation at all.

Reply:
Since the target PVC is also affected by the source-clone pod failure (maybe even more than the source), doesn't it also makes sense to add the annotations there? However, I see that the behavior will remain the same and we'll have fewer annotations cluttering up the PVC, so I'll commit the change.

Reply:
Everything is written to the target PVC, (source pod and target pod annotations). We should never update the source PVC in any of this. The datavolume controller only looks at the target PVC for the annotations.